openlibrary/plugins/openlibrary/connection.py

```diff
@@ -19,16 +22,31 @@
 class ConnectionMiddleware:
     response_type = "json"
 
-    def __init__(self, conn):
+    def __init__(
+        self, conn: Union[HybridConnection, MigrationMiddleware, MemcacheMiddleware]
```

Comment:
These classes are defined further down in this same file so delay their evaluation.
```suggestion
        self, conn: Union["HybridConnection", "MigrationMiddleware", "MemcacheMiddleware"]
```
Alternative solutions would be to move this class further down in the file or...
```
        self, conn: ConnectionMiddleware
```

Reply:
I think with these the quotes are the way to go, but for some reason they only seem to work with the `Union` syntax. If I try with a pipe, there is some error that I should probably investigate.